XIPs/xip-17-remote-attachment-content-type-proposal.md

```diff
@@ -0,0 +1,74 @@
+---
+xip: 17
+title: Remote Attachment Content Type
+description: Content type for remote attachments
+author: Pat Nakajima
+status: Review
+type: Standards Track
+category: XRC
+created: 2023-02-15
+---
+
+## Abstract
+
+This XRC proposes a new content type for messages that do not consist of text and/or are too large to be sent on the XMTP network. It builds on [XIP 15](https://github.com/xmtp/XIPs/pull/15)'s basic attachment type, but instead of messages being sent on the network, the message contains a URL. The contents of the URL is an encrypted protobuf encoded `EncodedContent` that can be decrypted and decoded by clients.
+
+## Motivation
+
+The current 1MB limit for messages on the XMTP network is too small for many types of content. To enable larger files, external storage is going to be necessary.
+
+## Specification
+
+Proposed content type:
+
+```js
+{
+  authorityId: "xmtp.org"
+  typeId: "remoteAttachment"
+  versionMajor: 0
+  versionMinor: 1
+}
+```
+
+The encoded content MUST have the following parameters:
+
+```js
+{
+	// The SHA256 hash of the remote content
+	checksum: string,
```

Comment:
checkSums are usually used in the context of data corruption. In this context the goal is message Integrity: Can we use "contentDigest" (or something similar) to reflect that?

Reply:
Yep!

Reply:
Is SHA256 an ok hash to use?

Reply:
Could be a good use-case for a CID from https://github.com/multiformats/js-multiformats (similar to what IPFS is using)

Reply:
Curious since I'm not super familiar with IPFS, what would that CIDs get us that SHA256 doesn't?

Reply:
Multihash/CID is kind of just an abstraction around different hashing methods for content. This is how IPFS docs describe it: https://docs.ipfs.tech/concepts/content-addressing/

![Screenshot 2023-02-21 at 2 23 42 PM](https://user-images.githubusercontent.com/182290/220439566-6f740340-4fb2-4cbc-a773-293cd8881983.png)